android/src/main/java/io/keen/client/android/AndroidJsonHandler.java

```diff
@@ -93,9 +137,25 @@ public void setWrapNestedMapsAndCollections(boolean value) {
      */
     private boolean isWrapNestedMapsAndCollections = (Build.VERSION.SDK_INT < 19);
 
+    /** Stringifier for converting JSONObjects to Strings; used for unit tests. */
+    private JsonObjectManager jsonObjectManager = null;
+
     ///// PRIVATE METHODS /////
 
     /**
+     * Get the default jsonObjectManager, or use one that was explicitly specified.
+     *
+     * @return A default implementation which uses the normal Android library methods, unless a
+     * different implementation has been set explicitly.
+     */
+    private JsonObjectManager getJsonObjectManager() {
```

Comment:
Hmm, I think that it should be safe as long as you don't have different threads trying to share the same `AndroidJsonHandler` instance but using different `JsonObjectManager` implementations. As long as they're both using the default `AndroidJsonObjectManager` (which should be the case for everything besides tests), the worst case would be that they race and accidentally instantiate multiple `AndroidJsonObjectManager`s - but since those are interchangeable I don't see a risk of any incorrect behavior.

I might be missing something, but I'm confident enough to ship it this way unless you have concerns (or can think of problematic scenarios).